packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -16,10 +18,17 @@ const BulkActionsBar = ({
 }) => {
   const { formatMessage } = useIntl();
   const { trackUsage } = useTracking();
+  const { data } = useSelector(listViewDomain());
 
   const [isConfirmButtonLoading, setIsConfirmButtonLoading] = useState(false);
   const [showConfirmDeleteAll, setShowConfirmDeleteAll] = useState(false);
 
+  const selectedEntriesObjects = data.filter((entry) => selectedEntries.includes(entry.id));
+  const showPublishButton =
+    showPublish && selectedEntriesObjects.some((entry) => !entry.publishedAt);
+  const showUnpublishButton =
+    showPublish && selectedEntriesObjects.some((entry) => entry.publishedAt);
```

Comment:
I guess there is a good amount of filtering going on here. The size of the array can't be bigger than 100 since it is tied to the pagination (unless that is configurable 🤔). I'm really not sure if we should wrap these in useMemo or not. @remidej wdyt?

Reply:
If we wanted to make it faster, I would recommend rewriting the logic so that we have a single loop (probably with reduce), instead of the loops of `.filter`, `.some` and `.some`.

It makes the code a bit more complex and not as readable though, so I would only do it if it brings a meaningful perf boost.

Reply:
IMO keep it as it is, if we have performance issues, _then_ we can optimise 👍🏼